core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -140,15 +150,50 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
 
         createdContainer match {
           case Some(((actor, data), containerState)) =>
+            //increment active count before storing in pool map
+            val (newData, container) = data match {
+              case p: PreWarmedData => //convert PreWarmedData -> WarmingData for concurrent cases
+                if (r.action.limits.concurrency.maxConcurrent > 1) {
+                  WarmingData(p.container, r.msg.user.namespace.name, r.action, Instant.now, 1) -> Some(p.container)
+                } else {
+                  p.copy(activeActivationCount = 1) -> Some(p.container)
+                }
+              case w: WarmedData =>
+                w.copy(activeActivationCount = w.activeActivationCount + 1) -> Some(w.container)
+              case pw: WarmingData =>
+                pw.copy(activeActivationCount = pw.activeActivationCount + 1) -> Some(pw.container)
+              case wnd: WarmingColdData =>
+                wnd.copy(activeActivationCount = wnd.activeActivationCount + 1) -> None
+              case n: NoData => //convert NoData -> WarmingColdData for concurrent cases
+                if (r.action.limits.concurrency.maxConcurrent > 1) {
+                  WarmingColdData(r.msg.user.namespace.name, r.action, Instant.now, 1) -> None
+                } else {
+                  n.copy(activeActivationCount = 1) -> None
+                }
+              case m: MemoryData => //convert MemoryData -> WarmingColdData for concurrent cases
+                if (r.action.limits.concurrency.maxConcurrent > 1) {
+                  WarmingColdData(r.msg.user.namespace.name, r.action, Instant.now, 1) -> None
+                } else {
+                  m.copy(activeActivationCount = 1) -> None
+                }
+            }
+
+            if (newData.activeActivationCount < 1) {
+              logging.error(this, s"invalid activation count < 1 ${newData}")
```

Comment:
Better to fail here (use `require`) as its more of an assertion failure indicating error in logic

Reply:
I'm reluctant to use `require()` here since in the unforeseen event of an error, the actor will restart, and be in some unknown state (e.g. previous ContainerProxy instances will be lost etc). I don't expect this to happen at this point, but I would rather delete the log message than add the require.